### ISSUE TITLE
X-Ray-like feature

### ISSUE THREAD

**Issue by boriar** [koreader/koreader#6023]
Is it possible to include support for X-Ray feature? Also, like is a feature for mobi and azw3, will be too difficult to implement also with epub?

Thanks for your work and attention

**Comment by Frenzie** [comment]
Seems to be a Kindle thing: https://en.wikipedia.org/wiki/X-Ray_(Amazon_Kindle)

I wouldn't hold my breath for new features being added to Mobi. ;-) More important, it seems like it has to be part of the file (?) so I imagine that's not a thing in EPUB.

If I understand that page correctly, the main thing effectively "missing" is the common knowledge about a book from a place like LibraryThing and/or GoodReads?

**Comment by boriar** [comment]
Yes. It's a feature that Amazon include with software for his Kindles. It isn't into/modified mobi but a separete archive with his own format and used only with mobi and azw3 format (not in new kfx).
I think is a usefull feature because inform about terms of a book, not a dictionary but like an encyclopedia. You know, show information about the characters or places named in the book.
Also, like you did sometimes with features from fb2 applies to epub, I ask if will be posible to implement on epub because is a more universal format.

You can inform about it better here (a thread on mobilread of a calibre plugin):
https://www.mobileread.com/forums/showthread.php?t=245754

And view an example here (Amazon presentation of the feature):
https://www.youtube.com/watch?v=AbzOLua2baw

**Comment by NiLuJe** [comment]
Someone cobbled something together by (ab)using `ruby` tags, which we don't support, IIRC.

There's already an issue open where something similar was disucussed in greater details ;).

**Comment by NiLuJe** [comment]
Oops, confused X-Ray w/ WordWise (https://github.com/koreader/koreader/issues/3044) ^^.

**Comment by xxyzz** [comment]
Hi @boriar, please try my calibre plugin: https://github.com/xxyzz/WordDumb. It adds X-Ray footnotes to EPUB books. I haven't release this feature yet, so you have to download the latest plugin from GitHub Actions(the artifact is double zipped) or build(zip) from source.

**Comment by boriar** [comment]
Hi @xxyzz , but I don't know how to compile the code from actions. If you show me (or link) a tutorial I will try it.

**Comment by xxyzz** [comment]
You can download the artifact from [GitHub Actions](https://github.com/xxyzz/WordDumb/actions/runs/1954771262), unzip then zip it, then install the zip file:

```
$ calibre-customize -a path_of_zip_file
```

Or clone the code, run these commands:

```
$ ./data/dump_lemmas.sh
$ calibre-customize -b .
```

**Comment by arpanghosh8453** [comment]
@xxyzz I have a suggestion for making this work! 

From what i understand, this will generate a separate modified file with the x-ray data. But I was wondering if we can design something that does not require modifying the original file. Because Koreader uses hash to idetify files across devices for syncing, it's hard to replace the whole library. Better if there is an open format (like a dictionary) where Koreader can search when the user highlight a name which is present in the x-ray. That way, the format will not cause issues and we don't need to modify the actual ebooks present. And with Koreader you will be able to support Kobo devices apart from kindle. 

I am not sure what the devs think about implementing this. Any thoughts @Frenzie and @NiLuJe? This seems like a good feature as it's popular in Kindle already!

**Comment by Frenzie** [comment]
For the most part that probably doesn't make too much of a difference, but if someone were interested it'd be a good thing to keep in mind.

**Comment by pazos** [comment]
Rejected here, see https://github.com/koreader/koreader/discussions/12964 for a 3rd party alternative
